engine-cdi/core/src/main/java/org/operaton/bpm/engine/cdi/impl/annotation/StartProcessInterceptor.java

```diff
@@ -63,8 +63,8 @@ public Object invoke(InvocationContext ctx) throws Exception {
       return result;
     } catch (InvocationTargetException e) {
       Throwable cause = e.getCause();
-      if(cause != null && cause instanceof Exception) {
-        throw (Exception) cause;
+      if (cause != null && cause instanceof Exception exception) {
```

Comment:
Just a question on how to proceed here: A user commented on a commit some time earlier, that `instanceof` is null safe. If the passed parameter is null, it is not an instance of `Exception` and the check is `false`, skipping the body. Is this something we should already improve here?

Reply:
See https://docs.oracle.com/javase/specs/jls/se17/html/jls-15.html#jls-15.20.2

> At run time, the result of the type comparison operator is determined as follows:
>
 >   If the value of the RelationalExpression is the null reference ([§4.1](https://docs.oracle.com/javase/specs/jls/se17/html/jls-4.html#jls-4.1)), then the result is false.

Reply:
the null check is obsolete then